usb/usb_host_hid/include/usb/hid_host.h

```diff
@@ -103,18 +120,10 @@ typedef struct {
     size_t task_priority;                   /**< Task priority of created background task */
     size_t stack_size;                      /**< Stack size of created background task */
     BaseType_t core_id;                     /**< Select core on which background task will run or tskNO_AFFINITY  */
-    hid_host_driver_event_cb_t callback;    /**< Callback invoked when HID driver event occurs. Must not be NULL. */
+    esp_event_handler_t callback;           /**< Callback invoked when HID driver event occurs. Must not be NULL. */
```

Comment:
@tore-espressif , 
`callback` here is the same as is used in the esp_hid_host example, [esp_hid_host_main.c, line 32](https://github.com/espressif/esp-idf/blob/76923a7a7f92ef81a2599d44c93c038143cc55f9/examples/bluetooth/esp_hid_host/main/esp_hid_host_main.c#L32)
With prototype: `void callback(void *handler_args, esp_event_base_t base, int32_t id, void *event_data)`.

The events are the same, except the ESP_HIDH_OPEN_EVENT (about what we agreed before). 

But probably, it can be eliminated during integration to the esp_hid....because the esp_hid already has the implementation. 
Anyway, it can be easily done. 

Do I need to add this changes to this MR also? (possibility to smoothly integration to the esp_hid)

Reply:
> Do I need to add these changes to this MR also?

If it makes sense to you, then yes. I always thought that the v2 is supposed to be ready to be integrated with `esp_hid`.

If you could provide an example of the integration with `esp_hid` (in this MR, or in different repo), it would help the reviewers with better understanding of the new API design, as well as with easier testing with real HID devices

Reply:
Could you please rebase this PR on top of current HID 1.0.1~1?

Reply:
Anyway, it is a good idea to make a rebase :+1: